components/PageHeader.vue

```diff
@@ -74,12 +78,17 @@
         return this.$i18n.locales.filter(i => i.code !== this.$i18n.locale);
       }
     },
+    watch: {
+      searchQuery: {
+        immediate: true,
+        handler(val) {
+          this.query = val ? val : '';
+        }
+      }
+    },
     methods: {
       submitSearchForm () {
-        this.$router.push(this.localePath({ name: 'search', query: { query: this.query ? this.query : '' } }));
-      },
-      getQueryFromParam () {
-        return this.$route.query ? this.$route.query.query : null;
+        this.$router.push(this.localePath({ name: 'search', query: { query: this.query } }));
       }
     },
     watchQuery: ['query']
```

Comment:
This can be removed as you noticed it's not doing anyting.

Reply:
Thought I had done; will do.